lib/chef/provider/package/snap.rb

```diff
@@ -356,6 +377,17 @@ def set_installed_package_conf(name, value)
           wait_for_completion(id)
         end
 
+        # Helper to construct error message
+        #
+        # @return [String] a formatted error message
+        def available_versions(elements)
```

Comment:
that seems to be very confusing to have `available_version` and `available_versions`, this should probably be named something else.